### DISCUSSION TITLE
Issue with I/O of RGBA Volumes

### DISCUSSION THREAD

**Discussion by c-langlet** [brainvisa/brainvisa.github.io#89] [Q&A] [answered]
Hello,

I have an issue with I/O of RGBA Volumes : I write a RGBA Volume in a png file, and when I read it back it is read as a Volume_U32. 
Any idea ? I used to do that and it was working. I changed some things in my code and doubled check but I don't think that I touched to this part, and anyway the small example following is done outside of the code. 

Example : 

```
>>> arr
array([[[[[ 20,  20,  20, 255]]]],
       [[[[  0,   0,   0, 255]]]],
       [[[[  0,   0,   0, 255]]]],
       ...,
       [[[[ 60,  60,  60, 255]]]],
       [[[[ 60,  60,  60, 255]]]],
       [[[[ 80,  80,  80, 255]]]]], dtype=uint8)
>>> vol = aims.Volume('RGBA', arr.shape[0])
>>> vol
<soma.aims.Volume_RGBA object at 0x7fada28b3ee8>
>>> vol['v'] = arr
>>> vol['v']
array([[[[[ 20,  20,  20, 255]]]],
       [[[[  0,   0,   0, 255]]]],
       [[[[  0,   0,   0, 255]]]],
       ...,
       [[[[ 60,  60,  60, 255]]]],
       [[[[ 60,  60,  60, 255]]]],
       [[[[ 80,  80,  80, 255]]]]], dtype=uint8)
>>> aims.write(vol, '/volatile/clanglet/tmp/test.png')
>>> vol2 = aims.read('/volatile/clanglet/tmp/test.png')
>>> vol2
<soma.aims.Volume_U32 object at 0x7facef16edc8>
>>> vol2['v']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/casa/host/build/python/soma/aims/__init__.py", line 984, in <lambda>
    numpy.asarray(self).__getitem__(*args, **kwargs)
IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
>>> vol2.np
array([[[[20]]],
       [[[ 0]]],
       [[[ 0]]],
       ...,
       [[[60]]],
       [[[60]]],
       [[[80]]]], dtype=uint32)

```

**Comment by c-langlet** [comment]
I found that I can use the dtype option inside the read method to force the reading as a Volume_RGBA, but I wasn't doing that previously and it was working, so it's puzzling me...

**Comment by denisri** [comment]
I see the same thing on your example, whatever version of aims I'm using (thus it does not seem to be new), however when I save a "real" image in PNG format, then it is read as RGBA. So it seems to more or less depend on the contents. I need to investigate more.

**Comment by denisri** [comment]
I opened brainvisa/aims-free#73

**Comment by denisri** [comment, marked as answer]
Fixed (partially) by brainvisa/aims-free@d6d8ca748964867b8c62ed8b4695abe34e5cb8b2.

**Reply by c-langlet** [reply]
I tested some configurations and it was indeed bugging when the color vectors were distributed uniformly (ex 0 0 0, 20 20 20, 40 40 40, etc)
